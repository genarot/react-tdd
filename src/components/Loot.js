import React from 'react';
import {connect} from "react-redux";
import {fetchBitCoin} from "../store/actions/bitcoin";

export const Loot = ({fetchBitCoin, bitcoin, balance}) => {
    React.useEffect(() => {
        // console.log(fetchBitCoin, bitcoin)
        fetchBitCoin();
    },[]);
    const computeBitcoin = () => {
        if ( Object.keys(bitcoin).length === 0 ) return ''
        else
            return balance / parseInt(bitcoin.bpi.USD.rate.replace(',',''), 10);
    }
    return (
        <div>
            <h3>Bitcoin balance: {computeBitcoin()}</h3>
        </div>
    );
};

const mapStateToProps = (state ) => ({
    ...state
});
const mapDispatchToProps = {
    fetchBitCoin
}
export default connect(mapStateToProps, mapDispatchToProps)(Loot);
