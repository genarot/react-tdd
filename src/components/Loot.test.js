import React from 'react';
import {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Loot} from "./Loot";

configure({adapter: new Adapter()});

describe('Loot', function () {
    let props = {balance: 10, bitcoin: {}};
    let loot = shallow(<Loot {...props}/>);

    it('renders properly', () => {
        expect(loot.debug()).toMatchSnapshot();
    });

    describe('when mounted', () => {
        const mockFetchBitcoin = jest.fn();
        beforeEach(() => {
            props.fetchBitCoin = mockFetchBitcoin;
            loot = mount(<Loot {...props}/>);
        });
        it('dispatches the "fetchBitcoin()" method ir receives from props ', function () {
            expect(mockFetchBitcoin).toHaveBeenCalled();
        });
    });

    describe('when there are valid bitcoin props', function () {
        beforeEach(() => {
            props = {balance: 10, bitcoin: {bpi: {USD: {rate: '1000'}}}};
            loot = shallow(<Loot {...props}/>);
        });
        it('display the correct bitcoin value', () => {
            expect(loot.find('h3').text()).toEqual('Bitcoin balance: 0.01')
        });
    });
});
