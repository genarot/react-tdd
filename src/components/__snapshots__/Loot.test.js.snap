// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renders properly 1`] = `
"<div>
  <h3>
    Bitcoin balance: 
  </h3>
</div>"
`;
